### PR TITLE
Supports UUID data type

### PR DESCRIPTION
**Pull request by EmadAlblueshi**
Fixes https://github.com/vietj/reactive-pg-client/issues/53

### CODE REVIEW

File: src/main/java/com/julienviet/pgclient/impl/codec/DataType.java [hunk 1]
```diff
@@ -435,10 +435,20 @@ public void encodeText(Buffer value, ByteBuf buff) {
   public static DataType<Object> MACADDR8 = new DataType<>(Object.class,774);
 
   // UUID
-  public static DataType<String> UUID = new DataType<String>(String.class, 2950) {
+  public static DataType<java.util.UUID> UUID = new DataType<java.util.UUID>(java.util.UUID.class, 2950) {
     @Override
-    public String decodeText(int len, ByteBuf buff) {
-      return buff.readCharSequence(len, StandardCharsets.UTF_8).toString();
+    public java.util.UUID decodeText(int len, ByteBuf buff) {
+      return java.util.UUID.fromString(buff.readCharSequence(len, StandardCharsets.UTF_8).toString());
```
Comment (vietj) [inline src/main/java/com/julienviet/pgclient/impl/codec/DataType.java]: shouldn't we use instead byte[] ?

**Comment by vietj** [comment]
it seems this PR reformats a lot of code

**Comment by vietj** [comment]
in the tests, we need to avoid that

File: src/main/java/com/julienviet/pgclient/impl/codec/DataType.java (hunk 1 above)
Comment (EmadAlblueshi) [reply src/main/java/com/julienviet/pgclient/impl/codec/DataType.java]: Here we are getting data as text

**Comment by EmadAlblueshi** [comment]
@vietj which test Binary or Text tests ?

**Comment by EmadAlblueshi** [comment]
@vietj ok I got it

File: src/main/java/com/julienviet/pgclient/impl/codec/DataType.java (hunk 1 above)
Comment (vietj) [reply src/main/java/com/julienviet/pgclient/impl/codec/DataType.java]: ah ok makes sense :-) and the other is using 2 longs which makes 16 bytes

Comment (vietj) [reply src/main/java/com/julienviet/pgclient/impl/codec/DataType.java]: and anyway it's not really going to be a bottleneck

Comment (EmadAlblueshi) [reply src/main/java/com/julienviet/pgclient/impl/codec/DataType.java]: Yes :)

**Comment by EmadAlblueshi** [comment]
@vietj is it :+1: now ?

**Comment by vietj** [comment]
much cleaner, I believe we need to refactor tests to remove boiler plate code soon so it remains maintainable

**Comment by EmadAlblueshi** [comment]
👍 Ok

**Pull request merged**